src/js/common/schemaform/FormApp.jsx

```diff
@@ -25,7 +25,9 @@ export default class FormApp extends React.Component {
   }
 
   onbeforeunload = e => {
-    const endpoint = this.props.currentLocation.pathname.split('/').pop();
+    const { currentLocation } = this.props;
+    const trimmedPathname = currentLocation.pathname.replace(/\/$/, '');
+    const endpoint = trimmedPathname.split('/').pop();
```

Comment:
Good idea @jbalboni, I've made this more consistent, and extracted this logic into a helper to appease CodeClimate.